### ISSUE TITLE
Remove the Axis range if there is no data associated to it

### ISSUE THREAD

**Issue by erinplourde** [cerner/carbon-graphs#312]
General graph logic:

* [x]  Graph attributes (axes, legends, regions...etc.)
* [ ]  Other (please specify): 

Specific graph logic:

* [ ]  Bar graph
* [ ]  Bubble graph
* [ ]  Gantt chart
* [x]  Line graph
* [ ]  Paired result
* [ ]  Pie chart
* [ ]  Scatter plot
* [ ]  Timeline

If there is no data present for all data sets on an axis, that axis range should be removed from the UI. 
This is applicable to both y1 and y2.  

Example of removing y1: 
<img width="1125" alt="Screen Shot 2020-10-19 at 1 35 23 PM" src="https://user-images.githubusercontent.com/52933964/96497209-f4743d00-120f-11eb-950c-92294ad9f94b.png">

Example of removing y2: 
<img width="903" alt="Screen Shot 2020-10-19 at 1 39 50 PM" src="https://user-images.githubusercontent.com/52933964/96497607-91cf7100-1210-11eb-9695-b20ca5f454e7.png">

**Comment by mraplumb** [comment]
@erinplourde  There are other graphs that include a Y2 axis that may have some of the same considerations here as the line graphs including Bars, Bubbles, Paired Results and Scatter plots.  

Holistically, we need to consider and decide how we handle Axis tics/values, Axis Labels, Axis Shapes and Legend items for each of these graph types when you pan to an area of the graphs that contains no values for a particular data set.  

In addition, we need to consider a "No Data" situation where there are no values at all for any data sets in the graphs related to the section of the graph in view.  There are concerns with showing our normal No Data graph layout for the fact that it might be misinterpreted to mean there is no data at all on the graph instead of just within this section.  Should we show an graph with the X and Y1 axis, no plotted values along with an arrow somewhere indicating which way to go to see the next available data points?

**Comment by erinplourde** [comment]
Current visuals for this issue :
Carbon Wiki Link: https://wiki.cerner.com/display/UserExperience/Carbon+Graphing+Library?preview=/2010049856/2412305431/Carbon_NoData.pdf
PDF: [Carbon_NoData.pdf](https://github.com/cerner/carbon-graphs/files/5731504/Carbon_NoData.pdf)
